src/Features/CSharp/Portable/Completion/CompletionProviders/ExplicitInterfaceMemberCompletionProvider.ItemGetter.cs

```
@@ -0,0 +1,280 @@
+
+                return items;
+
+                bool FilterInterfaceMember(ISymbol member)
+                {
+                    if (!member.IsAbstract && !member.IsVirtual)
+                        return false;
+
+                    if (member.IsAccessor() ||
+                        member.Kind == SymbolKind.NamedType ||
+                        !semanticModel.IsAccessible(node.SpanStart, member))
+                    {
+                        return false;
+                    }
+
+                    return true;
+                }
+
+                CompletionItem CreateCompletionItem(ISymbol member)
+                {
+                    var memberString = ToDisplayString(member, semanticModel);
+
+                    // Split the member string into two parts (generally the name, and the signature portion). We want
+                    // the split so that other features (like spell-checking), only look at the name portion.
+                    var (displayText, displayTextSuffix) = SplitMemberName(memberString);
+
+                    return MemberInsertionCompletionItem.Create(
+                        displayText, displayTextSuffix, DeclarationModifiers.None, line,
+                        member, targetToken, Position,
+                        rules: GetRules());
+                }
+            }
+            catch (Exception e)
+            when (FatalError.ReportAndCatchUnlessCanceled(e, ErrorSeverity.General))
+            {
+                // nop
+                return [];
+            }
+        }
+
+        private string ToDisplayString(ISymbol symbol, SemanticModel semanticModel)
+            => symbol switch
+            {
+                IEventSymbol eventSymbol => ToDisplayString(eventSymbol),
+                IPropertySymbol propertySymbol => ToDisplayString(propertySymbol, semanticModel),
+                IMethodSymbol methodSymbol => ToDisplayString(methodSymbol, semanticModel),
+                _ => "" // This shouldn't happen.
```

then throw.

---

Throwing instead of ignoring the unexpected case